src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -405,7 +409,18 @@ private static async Task<(string? idValue, string? partitionKeyValue)> GetIdAnd
                     if (string.Equals(item.Name.Value, "id", StringComparison.OrdinalIgnoreCase))
                     {
                         IList<ObjectFieldNode>? idValueObj = (IList<ObjectFieldNode>?)item.Value.Value;
-                        return idValueObj?.FirstOrDefault(x => x.Name.Value == "eq")?.Value?.Value?.ToString();
+
+                        ObjectFieldNode? itemToResolve = idValueObj?.FirstOrDefault(x => x.Name.Value == "eq");
```

Comment:
Can there be a different operator than `eq` here? like `gt` or `lt`

Reply:
I don't think we support  `gt` `lt` with partition key.